### PR TITLE
Pin the correct version of replicache-client in bindings build

### PR DESCRIPTION
**Pull request by aboodman**
Otherwise breaking changes in replicache-client have to be synchronously
fixed in bindings which isn't practical.

Related to https://github.com/rocicorp/replicache/issues/60

### CODE REVIEW

**Comment by aboodman** [comment]
@arv I made this for the bindings, but it's not working. It makes a lot of changes to sync_replay.json (is that expected?) but more importantly a test fails here:

```
00:08 +0 -1: sync [E]                                                                                                                                        
  Expected: <6>
    Actual: <9>
  
  package:test_api                                   expect
  package:flutter_test/src/widget_tester.dart 234:3  expect
  replicache_test.dart 577:7                         main.<fn>
```

halp?

**Comment by aboodman** [comment]
On the upside your unit test totally caught a compatibility break in repm so whee!

**Comment by arv** [comment]
```
  Expected: <6>
    Actual: <9>
```

Looks like it is replaying too much.

The json file changes a lot due to the hashes changing due to the commit contains a date. We might want to consider either having the test server return fake dates or having the test data redact the hashes. It also changes slightly due to the async nature but I think we could "fix" that by introducing some short Future.delayer in a few select places.

File: HACKING.md [hunk 1]
```diff
@@ -4,14 +4,39 @@ Run `./tool/build.sh` once. This will download the correct version of the Replic
...
+then re-run the tests:
 
 ```
-repl --db=/tmp/wherever serve
+flutter test
 ```
 
-Once you've done this, you can run the sample apps (in `sample` directory) and they will pick up Dart changes
-in the library as you make them.
+**Don't forget to undo the change to `replicache_test.dart` before landing!**
```
Comment (arv) [inline HACKING.md:39]: I'm doing this with an env variable in my current branch. Makes it cleaner

File: tool/build.sh [hunk 2]
```diff
@@ -2,18 +2,31 @@ ORIG=`pwd`
...
 mkdir build
 cd build
+
+# Copy the local repo into the output dir
 mkdir replicache-flutter-sdk
 ls ../ | grep -v build | grep -v sample | grep -v tool | grep -v gh-dl-release | grep -v doc | xargs -I{} cp -R ../{} replicache-flutter-sdk/{}
 
+# Build repm
+git clone https://github.com/rocicorp/replicache-client
+cd replicache-client
```
Comment (arv) [inline tool/build.sh:22]: [fyi] You could use `(` so you do not have to undo the `cd`

File: test/sync_replay.json [hunk 3]
```diff
@@ -2,20 +2,20 @@
...
     "method": "open",
-    "data": "{}",
+    "data": "null",
     "responseBody": ""
   },
   {
     "dbName": "sync",
     "method": "getRoot",
-    "data": "{}",
+    "data": "null",
```
Comment (arv) [inline test/sync_replay.json:11]: [fyi] I saw this. I changed the default to `const {}` again.

File: HACKING.md (hunk 1 above)
Comment (aboodman) [reply HACKING.md:39]: Thanks! Just update this when it lands.

File: tool/build.sh (hunk 2 above)
Comment (aboodman) [reply tool/build.sh:22]: I don't understand -- `(` ?

Comment (arv) [reply tool/build.sh:22]: Not a big deal. I was just pointing out that the shell script can be simplified slightly by doing something like:

```bash
(
  cd x
  doThing()
)
```

**Pull request merged**